libbeat/processors/script/javascript/javascript.go

```diff
@@ -0,0 +1,220 @@
+}
+
+// New constructs a new Javascript processor.
+func New(c *common.Config) (processors.Processor, error) {
+	conf := defaultConfig()
+	if err := c.Unpack(&conf); err != nil {
+		return nil, err
+	}
+
+	return NewFromConfig(conf, monitoring.Default)
+}
+
+// NewFromConfig constructs a new Javascript processor from the given config
+// object. It loads the sources, compiles them, and validates the entry point.
+func NewFromConfig(c Config, reg *monitoring.Registry) (processors.Processor, error) {
+	err := c.Validate()
+	if err != nil {
+		return nil, err
+	}
+
+	var sourceFile string
+	var sourceCode []byte
+
+	switch {
+	case c.Code != "":
+		sourceFile = "inline.js"
+		sourceCode = []byte(c.Code)
+	case c.File != "":
+		sourceFile, sourceCode, err = loadSources(c.File)
+	case len(c.Files) > 0:
+		sourceFile, sourceCode, err = loadSources(c.Files...)
+	}
+	if err != nil {
+		return nil, annotateError(c.ID, err)
+	}
+
+	s, err := newSession(sourceFile, sourceCode, c)
+	if err != nil {
+		return nil, annotateError(c.ID, err)
+	}
+
+	return &jsProcessor{
+		Config:     c,
+		s:          s,
+		sourceFile: sourceFile,
+		stats:      getStats(c.ID, reg),
+	}, nil
+}
+
+// loadSources loads javascript source from files.
+func loadSources(files ...string) (string, []byte, error) {
+	var sources []string
+	buf := new(bytes.Buffer)
+
+	readFile := func(path string) error {
+		if common.IsStrictPerms() {
+			if err := common.OwnerHasExclusiveWritePerms(path); err != nil {
+				return err
+			}
+		}
+
+		f, err := os.Open(path)
+		if err != nil {
+			return errors.Wrapf(err, "failed to open file %v", path)
+		}
+		defer f.Close()
+
+		if _, err = io.Copy(buf, f); err != nil {
+			return errors.Wrapf(err, "failed to read file %v", path)
+		}
+		return nil
+	}
+
+	for _, filePath := range files {
+		filePath = paths.Resolve(paths.Config, filePath)
+
+		if hasMeta(filePath) {
+			matches, err := filepath.Glob(filePath)
+			if err != nil {
+				return "", nil, err
+			}
+			sources = append(sources, matches...)
+		} else {
+			sources = append(sources, filePath)
+		}
+	}
+
+	if len(sources) == 0 {
+		return "", nil, errors.Errorf("no sources were found in %v",
+			strings.Join(files, ", "))
+	}
+
+	for _, name := range sources {
+		if err := readFile(name); err != nil {
+			return "", nil, err
+		}
+	}
+
+	return strings.Join(sources, ";"), buf.Bytes(), nil
+}
+
+func annotateError(id string, err error) error {
+	if err == nil {
+		return nil
+	}
+	if id != "" {
+		return errors.Wrapf(err, "failed in processor.javascript with id=%v", id)
+	}
+	return errors.Wrap(err, "failed in processor.javascript")
+}
+
+// Run executes the processor on the given it event. It invokes the
+// process function defined in the Javascript source.
+func (p *jsProcessor) Run(event *beat.Event) (*beat.Event, error) {
+	run := p.s.runProcessFunc
+	if p.stats != nil {
+		run = p.runWithStats
+	}
+	rtn, err := run(event)
+	return rtn, annotateError(p.ID, err)
+}
```

Comment:
We are actually opening the door for all kinds of scripts here, what comes to my mind are really long execution times or ReDoS cases, looking at the readme the VM support [interruption](https://github.com/dop251/goja#interrupting)

So I think we should set a default timeout and allow people to overrides that.

Reply:
Good idea.